### DISCUSSION TITLE
Storage size calculation overflowed while using Node2Vec

### DISCUSSION THREAD

**Discussion by ntquanghai** [pyg-team/pytorch_geometric#7396] [Q&A]
I currently have a homogeneous graph with these data:
`Data(edge_index=[2, 24386900], edge_attr=[24386900, 20])`

I intend to generate node embeddings via Node2Vec with these parameters
```
model = Node2Vec(data.edge_index, embedding_dim=64, walk_length=20,
             context_size=10, walks_per_node=10,
             num_negative_samples=1, p=1, q=1, sparse=True).to(device)
```

However, when I run this, it returns this error
`Storage size calculation overflowed with sizes=[9223291803303717676]`

I was wondering if there was any way to deal with this problem?

Thank you all in advance, and I would appreciate any response!

**Comment by rusty1s** [comment]
What's the output of `edge_index.max()`?
